ReportUpdater.php

```diff
@@ -22,97 +67,134 @@ public function __construct() {
 	public function updateReports( $config ) {
 		// Make sure config isn't empty
 		if ( !is_array( $config ) || count( $config ) < 1 ) {
-			logToFile( 'Error: Invalid config. Aborting!' );
+			wfLogToFile( 'Error: Invalid config. Aborting!' );
 			return;
 		}
-		foreach ( $config as $project => $info ) {
-			// Check that config values are set
-			if ( !isset( $info['Name'] ) || !isset( $info['Limit'] ) || !isset( $info['Report'] ) ) {
-				logToFile( 'Error: Incomplete data in config for ' . $project . '. Skipping.' );
-				continue;
-			}
-			// Make sure report will be written to Wikipedia namespace
-			if ( strpos( $info['Report'], 'Wikipedia:' ) !== 0 ) {
-				logToFile( 'Error: Invalid data in config for ' . $project . '. Skipping.' );
-				continue;
-			}
-			logToFile( 'Beginning to process: ' . $info['Name'] );
-			// Check the project exists
-			if ( !$this->api->doesTitleExist( $project ) ) {
-				logToFile( 'Error: Project page for ' . $info['Name'] . ' does not exist! Skipping.' );
-				continue;
-			}
-			$pages = $this->api->getProjectPages( $info['Name'] ); // Returns { 'title' => ['class'=>'', 'importance'=>''],...}
-			if ( empty( $pages ) ) {
-				continue;
-			}
-			// Get dates for last month
-			$start = strtotime( 'first day of previous month' );
-			$end = strtotime( 'last day of previous month' );
-			if ( count( $pages ) > 1000000 ) { // See T164178
-				logToFile( 'Error: ' . $project . ' is too large. Skipping.' );
+
+		foreach ( $config as $project => $projectConfig ) {
+			if ( false === $this->validateProjectConfig( $project, $projectConfig ) ) {
```

Comment:
This is a weird way to write a negation. How about just:
if ( !$this->validateProjectConfig( $project, $projectConfig ) ) {